src/Castle.Windsor/MicroKernel/DefaultKernel.cs

```diff
@@ -426,6 +426,31 @@ public virtual IHandler[] GetHandlers(Type service)
 			return result;
 		}
 
+		/// <summary>
+		///   Returns all handlers for all components
+		/// </summary>
+		/// <returns>Handler which is a sub dependency resolver for a component</returns>
+		public virtual IHandler[] GetHandlers()
+		{
+			var result = NamingSubSystem.GetAllHandlers();
+
+			// If a parent kernel exists, we merge both results
+			if (Parent != null)
+			{
+				var parentResult = Parent.GetHandlers();
+
+				if (parentResult.Length > 0)
+				{
+					var newResult = new IHandler[result.Length + parentResult.Length];
+					result.CopyTo(newResult, 0);
+					parentResult.CopyTo(newResult, result.Length);
+					result = newResult;
+				}
+			}
+
+			return result;
+		}
```

Comment:
@jnm2 - I have made [getting all handlers](https://github.com/castleproject/Windsor/pull/376/commits/463da9f141e8a0fa8c87d066679ae5e028995796#diff-ff15a9919978ddd9419b7185d67a319dR433) easy. This is going to start playing into [validation of configurations for tests](https://github.com/castleproject/Windsor/issues/328) if @jonorossi approves this PR.

I needed to validate against "framework configuration components" accidentally registered in the container here: https://github.com/castleproject/Windsor/pull/376/commits/463da9f141e8a0fa8c87d066679ae5e028995796